File: 0209 minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l=0,sum=0,le=Integer.MAX_VALUE;
        for(int r=0;r<nums.length;r++) {
            sum+=nums[r];
            while(sum>=target) {
                le=Math.min(r-l+1,le);
                sum-=nums[l++];
            }
        }
        return le==Integer.MAX_VALUE ? 0 : le;
    }
}

File: 0209 minimum-size-subarray-sum/solution.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int l=0,sum=0,m=int.MaxValue;
        for(int r=0;r<nums.Length;r++) {
            sum+=nums[r];
            while(sum>=target) {
                m=Math.Min(m,r-l+1);
                sum-=nums[l++];
            }
        }
        return m==int.MaxValue ? m=0 : m;
    }
}
